### PR TITLE
Switching from global api.Scheme to local Scheme

### PR DESCRIPTION
**Pull request by nilebox**
Fixes https://github.com/kubernetes-incubator/service-catalog/issues/1296

### CODE REVIEW

File: pkg/apis/servicecatalog/testapi/testapi.go [hunk 1]
```diff
@@ -30,62 +30,19 @@ import (
 	"reflect"
 	"strings"
 
+	"github.com/kubernetes-incubator/service-catalog/pkg/api"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer/recognizer"
-	"k8s.io/client-go/pkg/api"
```
Comment (MHBauer) [inline pkg/apis/servicecatalog/testapi/testapi.go]: not sure what this file is, but cool to get rid of code.

File: pkg/controller/controller_binding.go [hunk 2]
```diff
@@ -712,7 +712,7 @@ func (c *controller) injectServiceInstanceCredential(binding *v1alpha1.ServiceIn
 		err = nil
 
 		// Create new secret
-		secret := &v1.Secret{
+		secret := &apiv1.Secret{
```
Comment (MHBauer) [inline pkg/controller/controller_binding.go]: interesting that api and v1 are combined.

File: pkg/registry/servicecatalog/broker/storage.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ import (
 	"k8s.io/apiserver/pkg/registry/generic/registry"
 	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/apiserver/pkg/storage"
-	"k8s.io/client-go/pkg/api"
+	coreapi "k8s.io/client-go/pkg/api"
```
Comment (MHBauer) [inline pkg/registry/servicecatalog/broker/storage.go]: why this change? It does not look like we imported our 'api' here.

File: pkg/registry/servicecatalog/rest/storage_servicecatalog.go [hunk 4]
```diff
@@ -56,7 +56,7 @@ func (p StorageProvider) NewRESTStorage(
 		return nil, err
 	}
 
-	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(servicecatalog.GroupName, api.Registry, server.Scheme, server.ParameterCodec, server.Codecs)
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(servicecatalog.GroupName, api.Registry, api.Scheme, api.ParameterCodec, api.Codecs)
```
Comment (MHBauer) [inline pkg/registry/servicecatalog/rest/storage_servicecatalog.go]: need to check that this was removed from server or moved to api or see what happened.

File: pkg/api/register.go [hunk 5]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package server
```
Comment (MHBauer) [inline pkg/api/register.go]: this is the one I'm looking for at the bottom!

File: pkg/apis/servicecatalog/serialization_test.go [hunk 6]
```diff
@@ -270,38 +269,13 @@ func TestRoundTripTypes(t *testing.T) {
 	}
 }
 
-func testEncodePtr(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (nilebox) [reply pkg/apis/servicecatalog/serialization_test.go]: @MHBauer The main reason for removing this method as part of this PR was the snippet `... obj2.(*api.Pod);` - we shouldn't need/use core Kubernetes objects even in tests. Also this method is not being used, so I don't see why should we keep it.

File: pkg/apis/servicecatalog/testapi/testapi.go (hunk 1 above)
Comment (nilebox) [reply pkg/apis/servicecatalog/testapi/testapi.go]: @MHBauer we are mixing versioned and unversioned api packages, but in 1.8 both will be completely removed and only the versioned one is present in https://github.com/kubernetes/api/tree/master/core/v1
Hence the preparatory cleanup

File: pkg/controller/controller_binding.go (hunk 2 above)
Comment (nilebox) [reply pkg/controller/controller_binding.go]: @MHBauer I will rename all occurences of this package to the alias `corev1` while doing rebase onto 1.8 to be clear.

File: pkg/apis/servicecatalog/serialization_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (MHBauer) [reply pkg/apis/servicecatalog/serialization_test.go]: SGTM

**Comment by MHBauer** [comment]
@vaikas-google is that your own LGTM, or is it you adding a label for mine that I forgot?

File: pkg/registry/servicecatalog/broker/storage.go (hunk 3 above)
Comment (nilebox) [reply pkg/registry/servicecatalog/broker/storage.go]: @MHBauer 2 reasons:
1) I moved the scheme to our own `pkg/api` package and refer to it simply as `api` everywhere.
2) in 1.8 the package is moved to a separate repo https://github.com/kubernetes/api/tree/master/core/v1
see `core/v1`, hence the core alias prefix

File: pkg/registry/servicecatalog/rest/storage_servicecatalog.go (hunk 4 above)
Comment (nilebox) [reply pkg/registry/servicecatalog/rest/storage_servicecatalog.go]: @MHBauer yeah it was.

**Comment by nilebox** [comment]
> @vaikas-google is that your own LGTM, or is it you adding a label for mine that I forgot?

@MHBauer I like to think that it was Villes' own LGTM, and wait for yours ;)

File: pkg/apis/servicecatalog/testapi/testapi.go (hunk 1 above)
Comment (nilebox) [reply pkg/apis/servicecatalog/testapi/testapi.go]: re: getting rid of code
This file is used for serialization, fuzzer and rest client tests IIRC.
The code seem to be copy-pasted from main kube repo without thinking much - 90% of code was not used by our tests and was referencing core kube resources we don't need. explicitly left only our Service Catalog resources.

**Comment by vaikas-google** [comment]
Yes, that was my LGTM :)

**Comment by vaikas-google** [comment]
Yes, I agree on removing all this cruft. When I asked about it early on
(like why are we testing ods (iirc it was pods?)) it was not well
understood what all was necessary to be brought over.
